libraries/AP_Filesystem/AP_Filesystem_Sys.cpp

```diff
@@ -71,67 +67,46 @@ int AP_Filesystem_Sys::open(const char *fname, int flags)
         return -1;
     }
     struct rfile &r = file[idx];
-    r.data = new file_data;
-    if (r.data == nullptr) {
+    r.str = new ExpandingString;
+    if (r.str == nullptr) {
         errno = ENOMEM;
         return -1;
     }
 
     // This ensure that whenever new sys file is added its also added to list above
     int8_t pos = file_in_sysfs(fname);
-    uint32_t max_size = 0;
-    if (pos >= 0) {
-        max_size = sysfs_file_list[pos].filesize;
-    } else {
+    if (pos < 0) {
```

Comment:
You're leaking `r.str` here.  It's a preexisting issue.

Reply:
fixed